mainwindow.cpp

```diff
@@ -577,78 +513,96 @@ QStringList MainWindow::getAttackParameters()
     } else {
         // In case we do not have explicit format we erase remembered
         // key.
-        m_format = "";
+        m_format.clear();
     }
+    m_settings.setValue("formatJohn", m_format);
+    m_settings.setValue("formatUI", m_ui->comboBox_Format->currentText());
```

Comment:
As you can see there is some special treatment done there and I didn't want to do special treatment like this in the restoreSessionOptions method or make a useless method in MainWindow.
 if (m_ui->comboBox_Format->currentText() != tr("Auto detect")) {
        // We have one list for formats and subformats. Subformats
        // contain short description after it.
        // Strings could be like:
        // des
        // md5_gen(0): md5($p)  (raw-md5)
        // md5_gen(1001) md5(md5(md5(md5($p))))
        // User formats does not have semicolon between format name
        // and description. So we will restrict our pattern with right
        // brace.
        // NOTE: $ john -format:md5-gen -subformat:md5_gen(0)
        //       is deprecated. So we will use
        //       $ john -format:md5_gen(0)
        //       instead. See
        //       http://www.openwall.com/lists/john-users/2011/08/17/2
        // We remember format key to be used with '-show' to take
        // progress.
        m_format = "--format=" + m_ui->comboBox_Format->currentText();
        // Now we have '--format=format' or '--format=format(N)description'.
        // So we truncate string to ')' if brace is in string.
        //
        // We try to find ')'.
        int index = m_format.indexOf(")");
        // We check that we have brace in string.
        if (index >= 0) {
            // We truncate line to index keeping index.
            m_format.truncate(index + 1);
        }
        // We add format key onto parameters list.
        parameters << m_format;
    } else {
        // In case we do not have explicit format we erase remembered
        // key.
        m_format.clear();
    }